### PR TITLE
[FIX] sgd: Change deprecated n_iter to max_iter

### LINKED ISSUE #2917: SGDClassificationLearner deprecated parameter (n_iter)
DeprecationWarning: n_iter parameter is deprecated in 0.19 and will be removed in 0.21. Use max_iter and tol instead.

### PR DESCRIPTION
**Pull request by smiks**
##### Issue
SGDClassificationLearner deprecated parameter (n_iter). Fixes #2917.

##### Description of changes
Deprecation fix: changed n_iter to max_iter (SGDClassifier)

##### Includes
- [X] Code changes
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Orange/widgets/model/owsgd.py [hunk 1]
```diff
@@ -76,7 +76,7 @@ class Outputs(OWBaseLearner.Outputs):
     learning_rate_index = Setting(0)
     eta0 = Setting(.01)
     power_t = Setting(.25)
-    n_iter = Setting(5)
+    max_iter = Setting(10)
```
Comment (ajdapretnar) [inline Orange/widgets/model/owsgd.py]: Default in sklearn is 5. Is there a particular reason to have it at 10?

Comment (pavlin-policar) [inline Orange/widgets/model/owsgd.py]: Please implement a `migrate_settings` method. Without it, old workflows won't use the old `n_iter` value as the `max_iter` value while their semantics are essentially the same.

Comment (smiks) [reply Orange/widgets/model/owsgd.py]: No particular reason. Sklearn will change it to 1000 from 0.21. I decided for 10 only to see if I changed everything correctly (testing widget in Orange).

Comment (smiks) [reply Orange/widgets/model/owsgd.py]: I don't know how to do it.
Is this correct (added in class OWSGD)?

```
    @classmethod
    def migrate_settings(cls, settings_, version):
        if version < 2:
            settings_["max_iter"] = settings_.get("n_iter", 5)
```

Comment (pavlin-policar) [reply Orange/widgets/model/owsgd.py]: That should do the trick. Perhaps delete the setting `n_iter` as well to clean up? You can do this in one line `settings_["max_iter"] = settings_.pop("n_iter", 5)`.

**Comment by smiks** [comment]
I'll try to do both (```tol``` parameter and fix for SGDRegressionLearner).
I only have to fix learner (widget is ok - in case I don't add ```tol``` parameter), right?

**Comment by lanzagar** [comment]
Yes, the SGD widget calls the regression or classification learner depending on the type of data it gets.
It now already uses max_iter, so just the regression SGD class needs to be updated to match it.

**Comment by lanzagar** [comment]
It was hard to fall back to the old functionality (of setting just the number of iterations) without being able to set tol to None - I guess that is why you also removed migrate settings? I added a checkbox for `tol` and brought back migration.
You can double check my commit to make sure everything is still OK, then I think this is ready for merging.

**Comment by smiks** [comment]
I'll check it. It is my first PR for Orange so I still have a lot to learn about Orange.

**Comment by smiks** [comment]
Looks ok. I have also tested it locally with different datasets (iris and my datasets).  Tested different settings to see if both parameters work. Looks like it works, otherwise I wouldn't get this message ```Maximum number of iteration reached before convergence. Consider increasing max_iter to improve the fit.```

**Comment by smiks** [comment]
Thanks everyone. I'm happy to have contributed to Orange.

**Pull request merged**